src/figuresClasses.ts

```diff
@@ -1,11 +1,95 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  color: Figure['color'];
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: Figure['color'], a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('This figure does not have a negative value');
+    }
+
+    if (a + b <= c || b + c <= a || c + a <= b) {
+      throw new Error('This is not a triangle');
+    }
+
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: 'circle';
 
-export class Triangle implements Figure {}
+  color: Figure['color'];
 
-export class Circle implements Figure {}
+  radius: number;
 
-export class Rectangle implements Figure {}
+  constructor(color: Figure['color'], radius: number) {
+    if (radius <= 0) {
+      throw new Error('This figure does not have a negative value');
+    }
+
+    this.shape = 'circle';
+    this.color = color;
+    this.radius = radius;
+  }
+
+  getArea(): number {
+    const area = Math.PI * this.radius ** 2;
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape: 'rectangle';
```

Comment:
The `shape` property should be defined as a constant string 'rectangle' instead of using a type annotation. It should be `shape = 'rectangle';` to match the `Figure` interface.